### PR TITLE
Cleanup: use idiomatic Go.

### PR DESCRIPTION
**Pull request by bendory**
# Changes

There are no expected functional changes in this PR.

Cleanup: use idiomatic Go.

- Use context timeout rather than rolling our own.
- Use `time.Minute` rather than `60*time.Second`.
- In-lined single-use helper functions.
- Added comments with questions around unclear code.
- Use `t.Cleanup` instead of `defer` where appropriate.
- Return after `t.Error` when it's clear that continuing is pointless.

See:
- https://google.github.io/styleguide/go/decisions#matching-braces
- https://google.github.io/styleguide/go/decisions#cuddled-braces
- https://testing.googleblog.com/2017/06/code-health-reduce-nesting-reduce.html
- https://google.github.io/styleguide/go/best-practices#t-fatal
- https://pkg.go.dev/testing#T.Cleanup

/kind cleanup

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [N/A] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [N/A] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including
  functionality, content, code)
- [N/A] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings)
- [N/A] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

``` release-note
NONE
```

### CODE REVIEW

File: test/e2e_test.go [hunk 1]
```diff
@@ -155,19 +162,19 @@ func TestRekor(t *testing.T) {
...
 			// Setup the right config.
 			resetConfig := setConfigMap(ctx, t, c, test.cm)
-			defer resetConfig()
+			t.Cleanup(resetConfig)
 
 			createdObj := tekton.CreateObject(t, ctx, c.PipelineClient, test.getObject(ns))
 
 			// Give it a minute to complete.
-			waitForCondition(ctx, t, c.PipelineClient, createdObj, done, 60*time.Second)
+			waitForCondition(ctx, t, c.PipelineClient, createdObj, done, time.Minute)
```
Comment (lbernick) [inline test/e2e_test.go:174]: do these two lines also need nil checks? (here and below)

File: test/e2e_test.go [hunk 2]
```diff
@@ -251,31 +257,31 @@ func TestGCSStorage(t *testing.T) {
...
+	t.Cleanup(cleanup)
 
 	resetConfig := setConfigMap(ctx, t, c, map[string]string{
 		"artifacts.taskrun.signer":  "x509",
 		"artifacts.taskrun.storage": "gcs",
 		"storage.gcs.bucket":        bucketName,
 	})
-	defer resetConfig()
+	t.Cleanup(resetConfig)
 	time.Sleep(3 * time.Second)
```
Comment (lbernick) [inline test/e2e_test.go:274]: possibly another sleep that can be removed?

**Review by lbernick — APPROVED** [review]

**Comment by lbernick** [comment]
didn't realize this was a chains PR 😅 but you have my approval anyway, haha

File: test/e2e_test.go [hunk 3]
```diff
@@ -97,19 +97,26 @@ func TestTektonStorage(t *testing.T) {
...
-			defer resetConfig()
+			t.Cleanup(resetConfig)
 
 			createdObj := tekton.CreateObject(t, ctx, c.PipelineClient, test.getObject(ns))
 
 			// Give it a minute to complete.
-			waitForCondition(ctx, t, c.PipelineClient, createdObj, done, 60*time.Second)
+			if o := waitForCondition(ctx, t, c.PipelineClient, createdObj, done, time.Minute); o == nil {
+				t.Error("object never became `done`.")
+				return
```
[the remark this answers is not in the document]
Comment (bendory) [reply test/e2e_test.go:111]: The difference is that `t.Fatal` terminates to running `goroutine` while `t.Error...return` lets the caller continue.

In this case, there's no difference, and I think `t.Fatal` is more clear.

File: test/clients.go [hunk 4]
```diff
@@ -244,31 +251,10 @@ func setupSecret(ctx context.Context, t *testing.T, c kubernetes.Interface, opts
 	if _, err := c.CoreV1().Secrets(namespace).Update(ctx, &s, metav1.UpdateOptions{}); err != nil {
 		t.Error(err)
 	}
-	time.Sleep(60 * time.Second)
+	time.Sleep(time.Minute) // An explanatory comment would be nice here -- why do we sleep?
```
[the remark this answers is not in the document]
Comment (bendory) [reply test/clients.go:254]: https://github.com/tektoncd/chains/issues/664

File: test/e2e_test.go (hunk 1 above)
Comment (bendory) [reply test/e2e_test.go:174]: cleaned this up everywhere

File: test/e2e_test.go (hunk 2 above)
Comment (bendory) [reply test/e2e_test.go:274]: commented

**Review by lcarva — APPROVED** [review]
/lgtm

nice!

**Pull request merged**